src/commenter.js

```diff
@@ -0,0 +1,164 @@
+import Octokit from '@octokit/rest';
+import crypto from 'crypto';
+import { URL } from 'url';
+
+// Identify/parse and construct the deduplication comment that goes at the top of the PR comment.
```

Comment:
Really neat! Would be super cool for this to be made generic in some way… I wonder if `@octokit/rest` would be open to this as a feature idea for their commenting APIs.

Reply:
I think they'd probably want to try and fit this kind of functionality into their status checks API - which is a bit less convenient for us. I do think it'd be pretty cool for them to give us a mechanism to assign comments IDs because then we could generate an ID deterministically (like this) and PUT it to either create or update in a single call.

Reply:
Post on https://github.com/isaacs/github perhaps and/or email support@github.com (as that repo suggests) perhaps?